OmniBLE/Bluetooth/BluetoothManager.swift

```diff
@@ -54,10 +54,14 @@ class BluetoothManager: NSObject {
     }
     private let lockedStayConnected: Locked<Bool> = Locked(true)
 
```

Comment:
Maybe this is needed but I do worry about tracking transient states of the delegation process (i.e. disconnecting) at the class level since it relies on certain delegation events happening in an expected order... I've seen where these states can be the source of state bugs in similar cases where we get a stream of events like this. 

Ex: What if the peripheral doesn't permanently disconnect after this is set to true? Then isPermanentlyDisconnecting stays set to `true` which seems like a degenerate state.

This may be the best way to solve this but I just wanted to drop that consideration here in case that triggers any simple way to avoid this state.